snapcraft/internal/pluginhandler.py

```diff
@@ -650,15 +655,46 @@ def _migratable_filesets(fileset, srcdir):
     return snap_files, snap_dirs
 
 
-def _migrate_files(snap_files, snap_dirs, srcdir, dstdir, missing_ok=False,
-                   follow_symlinks=False, fixup_func=lambda *args: None):
+def _migrate_files_directories(snap_dirs, srcdir, dstdir,
+                               follow_symlinks=False, keep_owner=False):
     for directory in snap_dirs:
-        os.makedirs(os.path.join(dstdir, directory), exist_ok=True)
+        src = os.path.join(srcdir, directory)
+        dst = os.path.join(dstdir, directory)
+        os.makedirs(dst, exist_ok=True)
+        if keep_owner:
+            try:
+                st = os.stat(src, follow_symlinks=follow_symlinks)
+                uid = st.st_uid
+                gid = st.st_gid
+                os.chown(dst, uid, gid, follow_symlinks=follow_symlinks)
```

Comment:
Oh man, I forgot about that. Then by all means, do this all the time! That greatly simplifies things, no?